### PR TITLE
evidence: manage and initialize state objects more clearly in the pool

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: internal/consensus/byzantine_test.go [hunk 1]
```diff
@@ -87,10 +87,13 @@ func TestByzantinePrevoteEquivocation(t *testing.T) {
 				mempool.EnableTxsAvailable()
 			}
 
+			eventBus := eventbus.NewDefault(log.TestingLogger().With("module", "events"))
+			err = eventBus.Start(ctx)
+			require.NoError(t, err)
```
Comment (creachadair) [inline internal/consensus/byzantine_test.go:92]:
```suggestion
			require.NoError(t, eventBus.Start(ctx))
```

File: internal/consensus/byzantine_test.go [hunk 2]
```diff
@@ -327,69 +326,69 @@ func TestByzantineConflictingProposalsWithPartition(t *testing.T) {
 	// blocksSubs := make([]types.Subscription, n)
 	// reactors := make([]p2p.Reactor, n)
 	// for i := 0; i < n; i++ {
-	// 	// enable txs so we can create different proposals
-	// 	assertMempool(states[i].txNotifier).EnableTxsAvailable()
+	//	// enable txs so we can create different proposals
```
Comment (creachadair) [inline internal/consensus/byzantine_test.go:329]: I realize most of this is cosmetic changes to existing commented-out stuff, but could we get rid of the cmomented-out stuff? The test is already being skipped—if the issue is that the code doesn't even build, we are probably better off just deleting it. We can always recover the code from history.

File: internal/evidence/pool.go [hunk 3]
```diff
@@ -277,6 +250,31 @@ func (evpool *Pool) State() sm.State {
 	return evpool.state
 }
 
+func (evpool *Pool) Start(state sm.State) error {
+	if evpool.isStarted {
+		return errors.New("pooli is already running")
```
Comment (creachadair) [inline internal/evidence/pool.go:255]:
```suggestion
		return errors.New("pool is already running")
```

File: internal/evidence/pool_test.go [hunk 4]
```diff
@@ -34,6 +34,18 @@ var (
 	defaultEvidenceMaxBytes int64 = 1000
 )
 
+func bootstrapPool(t *testing.T, pool *evidence.Pool, store sm.Store) {
```
Comment (creachadair) [inline internal/evidence/pool_test.go:37]: Maybe: `startPool`?

**Review by creachadair — APPROVED** [review]

File: internal/consensus/byzantine_test.go (hunk 2 above)
Comment (tychoish) [reply internal/consensus/byzantine_test.go:328]: I'm inclined to do that in a different PR, but I'm definitely in support of this.

**Pull request merged**